pkg/services/publicdashboards/publicdashboard.go

```diff
@@ -14,31 +14,31 @@ import (
 
 //go:generate mockery --name Service --structname FakePublicDashboardService --inpackage --filename public_dashboard_service_mock.go
 type Service interface {
-	PublicDashboardEnabledExistsByAccessToken(ctx context.Context, accessToken string) (bool, error)
 	GetAnnotations(ctx context.Context, reqDTO AnnotationsQueryDTO, accessToken string) ([]AnnotationEvent, error)
 	GetDashboard(ctx context.Context, dashboardUid string) (*models.Dashboard, error)
 	GetMetricRequest(ctx context.Context, dashboard *models.Dashboard, publicDashboard *PublicDashboard, panelId int64, reqDTO PublicDashboardQueryDTO) (dtos.MetricRequest, error)
-	GetPublicDashboard(ctx context.Context, accessToken string) (*PublicDashboard, *models.Dashboard, error)
-	GetPublicDashboardConfig(ctx context.Context, orgId int64, dashboardUid string) (*PublicDashboard, error)
+	GetPublicDashboard(ctx context.Context, orgId int64, dashboardUid string) (*PublicDashboard, error)
+	GetPublicDashboardAndDashboard(ctx context.Context, accessToken string) (*PublicDashboard, *models.Dashboard, error)
 	GetPublicDashboardOrgId(ctx context.Context, accessToken string) (int64, error)
 	GetQueryDataResponse(ctx context.Context, skipCache bool, reqDTO PublicDashboardQueryDTO, panelId int64, accessToken string) (*backend.QueryDataResponse, error)
 	ListPublicDashboards(ctx context.Context, u *user.SignedInUser, orgId int64) ([]PublicDashboardListResponse, error)
-	PublicDashboardEnabled(ctx context.Context, dashboardUid string) (bool, error)
-	SavePublicDashboardConfig(ctx context.Context, u *user.SignedInUser, dto *SavePublicDashboardConfigDTO) (*PublicDashboard, error)
+	PublicDashboardEnabledExistsByAccessToken(ctx context.Context, accessToken string) (bool, error)
```

Comment:
Looking at our naming conventions- since go is strongly typed, the function signature will tell us what we need. I wonder if we need to bother is "ByAccessToken". We're not doing that anywhere else when we get by uid, etc.

Reply:
The problem with Go is it doesn't have function overloading so if you want to have 2 func to get a public dashboard, one by `uid` and one by `accessToken` this is necessary. I think we can omit it if we are getting a resource by its `id`  but I like to have it for other cases like `accessToken`.
It's also a common pattern for Repositories or Store layers.

Reply:
Sure- ok. I think that idiom is good then. add suffix if it is not by uid.

Ship it!